### PR TITLE
test: Modified index and slice tests in order to vary ranks

### PR DESCRIPTION
**Pull request by DeaMariaLeon**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [ ] 🪄 Feature
- [ ] 🐞 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📚 Documentation
- [x] 🧪 Test
- [ ] 🛠️ Other

## Related issues

- Related issue #
- Closes #

## Checklist

- [ ] Code follows style guide
- [x] Tests added
- [ ] Documented the changes

***

## Please explain your changes below.
Original benchmarks used fixed rank

### CODE REVIEW

File: benchmarks/test_benchmark_coo.py [hunk 1]
```diff
@@ -29,15 +29,15 @@ def bench():
         x @ y
 
 
-def elemwise_test_name(param):
+def id_of_test(param):
```
Comment (hameerabbasi) [inline benchmarks/test_benchmark_coo.py:32]: I'd recommend naming this `get_test_id`.

File: benchmarks/test_benchmark_coo.py [hunk 2]
```diff
@@ -77,65 +77,47 @@ def bench():
...
+        x[(slice(side // 2),) * rank]
 
 
 def test_index_fancy(benchmark, indexing_args, seed):
     x = indexing_args
     side = x.shape[0]
+    rank = x.ndim
     rng = np.random.default_rng(seed=seed)
-    index = rng.integers(0, side, side // 2)
+    index = rng.integers((side // 2,) * rank)
```
Comment (hameerabbasi) [inline benchmarks/test_benchmark_coo.py:120]:
```suggestion
    index = rng.integers(0, side, size=(side // 2,) * rank)
```
The [signature for `rng.integers`](https://numpy.org/doc/stable/reference/random/generated/numpy.random.Generator.integers.html) is `(min, max, size)`.

**Review by hameerabbasi — CHANGES_REQUESTED** [review]
Overall, this looks excellent. Thanks for the work, @DeaMariaLeon! I've left some minor comments.

File: benchmarks/test_benchmark_coo.py (hunk 1 above)
Comment (DeaMariaLeon) [reply benchmarks/test_benchmark_coo.py]: Done

File: benchmarks/test_benchmark_coo.py (hunk 2 above)
Comment (DeaMariaLeon) [reply benchmarks/test_benchmark_coo.py]: Corrected

**Review by hameerabbasi — APPROVED** [review]

File: benchmarks/test_benchmark_coo.py [hunk 3]
```diff
@@ -77,65 +77,47 @@ def bench():
...
+        x[(slice(side // 2),) * rank]
 
 
 def test_index_fancy(benchmark, indexing_args, seed):
     x = indexing_args
     side = x.shape[0]
+    rank = x.ndim
     rng = np.random.default_rng(seed=seed)
-    index = rng.integers(0, side, side // 2)
+    index = rng.integers(0, side, size=(side // 2,) * rank)
```
Comment (hameerabbasi) [inline benchmarks/test_benchmark_coo.py:120]: This issue is here -- we need to make this array 1D as `sparse` doesn't yet support N-dimensional array indices. We can remove the ` * rank` here.

Comment (DeaMariaLeon) [reply benchmarks/test_benchmark_coo.py]: Done. Thanks.

**Review by hameerabbasi — APPROVED** [review]

**Comment by DeaMariaLeon** [comment]
I tried to "acknowledge the performance issues" on codspeed, but I'm lacking the rights to do it. 
I'm talking about the report's note above:

> ⚠️ Please fix the performance issues or [acknowledge them on CodSpeed](https://codspeed.io/pydata/sparse/branches/DeaMariaLeon:bench3).

It's just that the benchmark changed name, and codspeed "thinks" that it's missing. I think we need to fix this in order to merge the PR, as codspeed will block. @hameerabbasi

**Pull request merged**